### ISSUE TITLE
Pre-approval of failed tests on CQ doesn't work

### ISSUE THREAD

**Issue by alexmarkov** [dart-lang/sdk#36015]
Trying to follow the new pre-approval workflow on https://dart-review.googlesource.com/c/sdk/+/93989/2.

```
$ tools/sdks/dart-sdk/bin/dart tools/approve_results.dart -p https://dart-review.googlesource.com/c/sdk/+/93989/2 -b dart2js-*
Selected bot: dart2js-strong-linux-x64-chrome
Selected bot: dart2js-strong-hostasserts-linux-ia32-d8
Selected bot: dart2js-minified-strong-linux-x64-d8
Selected bot: dart2js-unit-linux-x64-release
Unhandled exception:
Exception: The log at https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8920580854075409568/+/steps/gsutil_find_latest_build/0/logs/raw_io.output_text_latest_/0?format=raw doesn't exist
#0      loadLog (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:86:5)
<asynchronous suspension>
#1      loadResultsFromBot (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:110:21)
<asynchronous suspension>
#2      main (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:469:14)
<asynchronous suspension>
#3      _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#4      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

```
$ tools/sdks/dart-sdk/bin/dart tools/approve_results.dart -p https://dart-review.googlesource.com/c/sdk/+/93989/2 -b dart2js-strong-linux-x64-chrome
^[Selected bot: dart2js-strong-linux-x64-chrome
Unhandled exception:
Exception: The log at https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8920580854075409568/+/steps/gsutil_find_latest_build/0/logs/raw_io.output_text_latest_/0?format=raw doesn't exist
#0      loadLog (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:86:5)
<asynchronous suspension>
#1      loadResultsFromBot (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:110:21)
<asynchronous suspension>
#2      main (file:///Users/alexmarkov/dart/sdk/tools/approve_results.dart:469:14)
<asynchronous suspension>
#3      _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#4      _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)
```

/cc @whesse @athomas

**Comment by whesse** [comment]
My fault.  I moved the step that find the latest results, and didn't change
its name here.  Will fix now.

**Comment by athomas** [comment]
Fix landed in 944544137b00cf23a5f3fd27b00faca5c2373507.

**Comment by sortie** [comment]
We've put in a temporary workaround that supports old and new try runs. The workaround will be removed after a month has passed. Pre-approval will work correctly if you have the fix in 03ee0fea5182a83a7521ed9c515615695c90141c. If you're prior to that commit, you will be able to approve try runs made before the recipe changed.

**Comment by karlklose** [comment]
Removed workaround in cde9ff36e53c53d3cf6a0b859b1c241f053f8349
